src/order/monotone.lean

```diff
@@ -462,6 +462,25 @@ lemma strict_anti.comp_strict_mono_on (hg : strict_anti g) (hf : strict_mono_on
   strict_anti_on (g ∘ f) s :=
 λ a ha b hb h, hg (hf ha hb h)
 
+namespace list
+
+theorem foldl_monotone {f : α → β → α} (H : ∀ b, monotone (λ a, f a b)) (l : list β) :
+  monotone (λ a, l.foldl f a) :=
+list.rec_on l (λ _ _, id) (λ i l hl _ _ h, hl (H _ h))
+
+theorem foldr_monotone {f : α → β → β} (H : ∀ a, monotone (f a)) (l : list α) :
+  monotone (λ b, l.foldr f b) :=
+λ _ _ h, list.rec_on l h (λ i l hl, H i hl)
+
+theorem foldl_strict_mono {f : α → β → α} (H : ∀ b, strict_mono (λ a, f a b)) (l : list β) :
+  strict_mono (λ a, l.foldl f a) :=
+list.rec_on l (λ _ _, id) (λ i l hl _ _ h, hl (H _ h))
+
+theorem foldr_strict_mono {f : α → β → β} (H : ∀ a, strict_mono (f a)) (l : list α) :
+  strict_mono (λ b, l.foldr f b) :=
+λ _ _ h, list.rec_on l h (λ i l hl, H i hl)
+
+end list
 end composition
```

Comment:
shitespace nit:
```suggestion
end list

end composition
```
Just to check: these are deliberately under `composition` because `fold` is just iterated composition?

Reply:
Yep, that's where I thought they fit best.